kyaml/sets/sets.go

```diff
@@ -3,6 +3,37 @@
 
 package sets
 
+type StringList [][]string
```

Comment:
For this to be in this package, it has to be a set of string lists.

But it doesn't act like a set, e.g. one can append the same list twice and it will appear twice.

If that's what we're shooting for, and its important to the rest of the PR, 
please
 - add comment  above the `type` definition showing some examples (i was confused
   by the name and had to read the code, which implies we need a comment).
 - add a test file (sets_test.go) and a test which tests ithat inserting the same thing twice
   doesn't change the size of the set
 - put that in a different PR to get the size of this PR down.

Reply:
opened #3173, thanks!

Reply:
now merged, thanks